juicy-select.html

```diff
@@ -42,17 +42,25 @@
                 this.$select.addEventListener("change", this.selectChange.bind(this));
                 this.set("multiple", this.$select.hasAttribute("multiple"));
 
+                this.isAttached = true;
+
                 setTimeout(function () {
+                    if (!this.isAttached)
```

Comment:
What is the point of this check, if the property was set to `true` a few lines above? Is it because the element may get detached from the DOM during those 10 ms of timeout?

Reply:
Yes, that's exactly why! It's not just in theory. It actually happened!

Reply:
Then it's a bad way to solve this situation. Instead it should be:

```js
this._initializationTimeout = setTimeout(function () { ... });
```

```js
detached: function () {
    clearTimeout(this._initializationTimeout);
}
```

Reply:
Obviously! 👍